frontend/src/pages/casedesigner/detail/BookcaseDetail.tsx

```diff
@@ -0,0 +1,314 @@
+import BookcaseDetailTable from "./BookcaseDetailTable";
+import TextLabel from "../../../components/text/TextLabels";
+import { TextEditor } from "../../../components/editors/TextEditor";
+import { NumberEditor } from "../../../components/editors/NumberEditor";
+
+const emptyBookcase: Bookcase = {
+  id: "",
+  name: "",
+  width: 0,
+  creator: "",
+  lastEditDate: new Date(),
+  lastEditor: "",
+  shelves: [],
+};
+
+const emptyShelf: Shelf = {
+  id: "",
+  displayedBooks: [],
+};
+
+export default function BookcaseDetail() {
+  // -------- STATE --------
+  // From URL
+  const { id } = useParams();
+  const isAddPage = id === undefined;
+  const [isModifiable, setIsModifiable] = useState<boolean>(id === undefined);
+
+  // The navigator to switch pages
+  const navigate = useNavigate();
+
+  const [originalData, setOriginalData] = useState<Bookcase>(emptyBookcase);
+
+  // useImmer is used to set state for nested data in a simplified format
+  const [bookcase, setBookcase] = useImmer<Bookcase>(emptyBookcase);
+  const [isConfirmationPopupVisible, setIsConfirmationPopupVisible] =
+    useState<boolean>(false);
+  const [deletePopupVisible, setDeletePopupVisible] = useState<boolean>(false);
+  const [isGoBackActive, setIsGoBackActive] = useState<boolean>(false);
+
+  // Load the Bookcase data on page load
+  useEffect(() => {
+    if (!isAddPage) {
+      CASE_DESIGNER_API.getBookcaseDetail(id)
+        .then((response) => {
+          const bookcase = APIToInternalBookcaseConversion(response);
+          setOriginalData(bookcase);
+          setBookcase(bookcase);
+        })
+        .catch(() => showFailure(toast, "Could not fetch bookcase data"));
+    }
+  }, []);
+
+  // Call to delete the bookcase
+  const deleteBookcaseFinal = () => {
+    logger.debug("Delete Bookcase Finalized");
+    setDeletePopupVisible(false);
+    CASE_DESIGNER_API.deleteBookcase(id!)
+      .then(() => {
+        showSuccess(toast, "Bookcase deleted");
+        navigate("/bookcase");
+      })
+      .catch(() => showFailure(toast, "Failed to delete bookcase"));
+  };
+
+  const onSubmit = (): void => {
+    if (isAddPage) {
+      callAddBookcaseAPI();
+    } else {
+      callModifyBookcaseAPI();
+    }
+  };
+
+  const callAddBookcaseAPI = () => {
+    const APIbookcase = InternalToAPIBookcaseConversion(bookcase);
+    CASE_DESIGNER_API.addBookcase(APIbookcase)
+      .then(() => {
+        showSuccess(toast, "Bookcase added successfully");
+        isGoBackActive ? navigate("/bookcases") : undefined;
+      })
+      .catch((error) => {
+        showFailure(toast, error.data.errors[0] ?? "Failed to add bookcase");
+      });
+  };
+
+  const callModifyBookcaseAPI = () => {
+    const APIbookcase = InternalToAPIBookcaseConversion(bookcase);
+    CASE_DESIGNER_API.modifyBookcase(APIbookcase)
+      .then(() => {
+        showSuccess(toast, "Bookcase modified successfully");
+        isGoBackActive ? navigate("/bookcases") : undefined;
```

Comment:
For the undefined Element here this should update your original data store with the modified book data. This ensure the next edit -> cancel works properly after a submit. Also for the submit, none of the books added to the table showed up at all despite your API Get request having them.

Reply:
Yup, I made a ticket for this, the data flow isn't fixed yet, I just needed to stop the page from reloading so I could test a couple things out.